vendor/zkllvm-metacraft-circuits/src/circuit_utils/circuit_byte_utils.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "base_types.h"
+
+using namespace nil::crypto3;
+
+namespace circuit_byte_utils {
+
+    using sha256_t = typename hashes::sha2<256>::block_type;
+
+    bool is_same(sha256_t block0, sha256_t block1) {
+
+        bool result = true;
+        for (auto i = 0; i < sizeof(block0) / sizeof(block0[0]) && result; i++) {
+            printf("Element found %d\n", i);
+            result = result && (block0[0] == block1[0]);
+        }
+
+        return result;
+    }
+
+    template<typename T>
+    char get_nth_byte(const T& val, unsigned int n) {
+        static_assert(std::is_integral<typename std::remove_reference<T>::type>::value, "T must be integral");
+        assert_true(n < sizeof(T));
```

Comment:
Consider removing this assert

Reply:
Might be keep it just to be on the safe side.